### ISSUE TITLE
[linux] Assert doesn't work on Linux

### ISSUE THREAD

**Issue by qiaojingx** [01org/zephyr.js#1687]
### Description

This issue a regression issue. Most of tests in zephyr.js will get TypeError on Linux.
 
### Test Code

[test-buffer.js](https://github.com/01org/zephyr.js/blob/master/tests/test-buffer.js);
[test-buffer-rw.js](https://github.com/01org/zephyr.js/blob/master/tests/test-buffer-rw.js);
[test-callbacks.js](https://github.com/01org/zephyr.js/blob/master/tests/test-callbacks.js);
[test-timers.js](https://github.com/01org/zephyr.js/blob/master/tests/test-timers.js);
[test-error.js](https://github.com/01org/zephyr.js/blob/master/tests/test-error.js);
[test-eval.js](https://github.com/01org/zephyr.js/blob/master/tests/test-eval.js);
[test-event.js](https://github.com/01org/zephyr.js/blob/master/tests/test-event.js);
[test-gpio.js](https://github.com/01org/zephyr.js/blob/master/tests/test-gpio.js);
[test-ocfserver.js](https://github.com/01org/zephyr.js/blob/master/tests/test-ocfserver.js);
[test-ocfclient.js](https://github.com/01org/zephyr.js/blob/master/tests/test-ocfclient.js);
[test-board.js](https://github.com/01org/zephyr.js/blob/master/tests/test-board.js);

### Steps to Reproduction

1. `make BOARD=linux`
2. `./outdir/linux/release/jslinux tests/test-buffer.js`
 
or simply code:
```
var assert = require('Assert.js');
assert(true, "assert works fine");
```
### Actual Result

![selection_008](https://user-images.githubusercontent.com/19813546/33539166-ef6974ac-d8ff-11e7-8187-535b7e3296a6.png)

 ### Expected Result

![selection_010](https://user-images.githubusercontent.com/19813546/33539219-38b728a2-d900-11e7-9724-d4ee701a5d05.png)

### Test Builds

Branch | Commit Id | Target Device | Test Date | Result
---|---|---|---|---
master | cbec79d | Arduino 101 | Dec 4, 2017 | Pass
master | 091f6ce - cbec79d | Linux | Dec 4, 2017 | Fail
master | 091f6ce - cbec79d | MacOS | Dec 4, 2017 | Fail
master | 62697b3 | Linux | Dec 4, 2017 | Pass
master | 62697b3 | MacOS | Dec 4, 2017 | Pass

**Comment by jimmy-huang** [comment]
This was broken by the Modules patch, @brianjjones please take a look at it.

**Comment by brianjjones** [comment]
Made a fix for this #1691

**Comment by qiaojingx** [comment]
Verified with 79cd6aa from https://github.com/brianjjones/zephyr.js-1/tree/modules-linux. This issue can be fixed. 
I will close this bug after patch merged.
